docs/runtimes-building.md

```diff
@@ -0,0 +1,146 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more contributor
+# license agreements.  See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.  The ASF licenses this file to you
+# under the Apache License, Version 2.0 (the # "License"); you may not use this
+# file except in compliance with the License.  You may obtain a copy of the License
+# at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+-->
+
+# Working with Runtimes locally
+
+This document is designed to be applied to the various
+`incubator-openwhisk-runtimes-*` repositories as they are converted to
+multi-architecture build. The multi-architecture build supports building Docker
+multi-architecture manifest lists across (currently) amd64, Power 64
+(little-endian), and System Z (s390x) architectures.  The framework is designed
+to extend to other chips, including the Arm64 chipset.
+
+You can tell that a runtime has been converted to a multi-architecture build
+by referring to its root `README.md` file.
+
+## Reading this document
+
+For this document, we will assume an environment variable `ACTION_NAME` has
+been set to the directory name of an action in the runtime repository on which
+work is being done (i.e. set to `nodejs6Action` while working in
+`incubator-openwhisk-runtime-nodejs`).
+
+## Building a single-architecture Docker image locally
+
+To build a single-architecture image locally on a 64-bit x86 machine, execute
+
+```
+./gradlew core:$ACTION_NAME:dockerBuildImage -Pdocker_local_json='{"amd64":null}'
+```
+
+To check on what images will be produced, look in the `settings.json` file in
+the runtime repository (look for `gradle.ext.dockerBuildProjects =`).  The
+default image prefix is currently 'whisk'.
+
+## Building a multi-architecture Docker image
+
+Multi-architecture builds work be repeating build steps on independent docker
+instances for each architecture. To ensure security, it's necessary to set up
+the docker instances to use TLS and to have the TLS client certificates (
+`ca.pem`, `cert.pem` and `key.pem`) available in some directory on the local
```

Comment:
do you use the certs that are generated for the openwhisk deployment or new ones?

Reply:
New certs.  As a separate TODO, I’m trying to automate cert generation so it won’t be a 15-minute process to create the certs — but that’s a nice-to-have right now.

Reply:
It could be done with existing certs, if your SSL/TLS-fu is up to it.  It's just a matter of knowing which cert to put where and how to configure the Docker daemon.